### PR TITLE
Split audit.rules into multiple files

### PR DESCRIPTION
**Pull request by frederikbosch**
I have done four things.

1. audit.rules split into three files
2. added a header and footer audit file
3. changed the configuration to allow a space separated list of audit rules
4. concatenate header, configuration rules and footer to a single audit.rules file

### CODE REVIEW

File: scripts/22_auditd [hunk 1]
```diff
@@ -6,7 +6,13 @@ function f_auditd {
   sed -i 's/^max_log_file_action =.*/max_log_file_action = keep_logs/' "$AUDITDCONF"
   sed -i 's/^space_left_action =.*/space_left_action = email/' "$AUDITDCONF"
   sed -i 's/^GRUB_CMDLINE_LINUX=.*/GRUB_CMDLINE_LINUX="audit=1"/' "$DEFAULTGRUB"
-  cp "$AUDITD_RULES" /etc/audit/audit.rules
+
+  cp "./misc/audit.header" /etc/audit/audit.rules
```
Comment (konstruktoid) [inline scripts/22_auditd]: This is static links, and `cp /etc/audit/audit.rules "$AUDITRULES"` is already sorted on line 17.

File: scripts/22_auditd [hunk 2]
```diff
@@ -6,7 +6,13 @@ function f_auditd {
...
+  for l in $AUDITD_RULES; do
```
Comment (konstruktoid) [inline scripts/22_auditd]: `while read -r` is preferred.

**Comment by frederikbosch** [comment]
@konstruktoid Thanks for the review. What is your general thought of the suggested changes?

**Comment by frederikbosch** [comment]
I have added two more changes.
1. Removed the watch rules for tmpfs. This not necessary any more now these filesystems are mounted with noexec.
2. Added an audit backlog limit to the boot parameters in order to prevent lost messages.

**Comment by konstruktoid** [comment]
Yeah, nice :)

File: scripts/22_auditd (hunk 2 above)
Comment (frederikbosch) [reply scripts/22_auditd]: Isn't that for a line by line split? I am not getting it done for the space separated variable `$AUDITRULES`.

File: scripts/22_auditd (hunk 1 above)
Comment (frederikbosch) [reply scripts/22_auditd]: The double `cp` is intentional as the current script also does that, [line 9](https://github.com/konstruktoid/hardening/blob/master/scripts/22_auditd#L9) and [line 11](https://github.com/konstruktoid/hardening/blob/master/scripts/22_auditd#L11). I believe this is done to let auditctl work with the new configuration directly. Alternatively we could call augenrules.

Regarding the static links for the header and footer. By making those two files static, they become the ones that are really required. What would you suggest?

File: scripts/22_auditd (hunk 2 above)
Comment (konstruktoid) [reply scripts/22_auditd]: Yeah you're right. Ignore my comment,

File: scripts/22_auditd (hunk 1 above)
Comment (frederikbosch) [reply scripts/22_auditd]: So if we agree on this part of the PR, you might be able to merge it.

**Comment by konstruktoid** [comment]
Thanks!

**Pull request merged**